README.md

```diff
@@ -13,30 +17,30 @@ before:
 - Your API is returning URLs for your data's images.
 - Those images are sizable and take a bit to download.
 - The image URLs are stable:
-  - URL uniquely identifiers an image.
+  - URL uniquely identifies an image.
   - Image pointed at by a given URL will not change.
 - Download may fail because network or because backend.
 - You want to display the UI already and update the images when they arrive.
 
 This library won't help you with displaying _good_ UI while you wait for image downloads. You better convince your
-backend workmates to send in some media metadata like the image size. But as for a reasonably efficient caching system
-for the images you would be writing something like _this_:
+backend workmates to send in some media metadata like the image size. But as for a reasonably efficient fetch and cache
+system for the images you could be writing something like _this_:
 
 ```swift
-import SwiftCache
+import ResourceProvider
 
 struct ImageConversionError: Error {}
 
-func buildImageCache() -> AnyThrowingAsyncCache<URL, UIImage> {
-    Cache.networkDataSource()
+func buildImageProvider() -> ThrowingAsyncProvider<URL, UIImage> {
```

Comment:
May want to consider `makeImageProvider` here following the recommendations for factory functions: https://www.swift.org/documentation/api-design-guidelines/#strive-for-fluent-usage

Reply:
Thanks for pointing that out.